modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java

```diff
@@ -218,42 +191,28 @@
   /**
    * Creates a new instance of the streaming distribution service.
    */
-  public WowzaAdaptiveStreamingDistributionService() {
+  public WowzaStreamingDistributionService() {
     super(JOB_TYPE);
   }
 
-  public void activate(BundleContext bundleContext) {
-    // Get the configured streaming and server URLs
-    if (bundleContext != null) {
+  public String getDistributionType() {
+    return DISTRIBUTION_TYPE;
+  }
 
-      String readStreamingUrl = StringUtils.trimToNull(bundleContext.getProperty(STREAMING_URL_KEY));
-      String readStreamingPort = StringUtils.trimToNull(bundleContext.getProperty(STREAMING_PORT_KEY));
-      String readAdaptiveStreamingUrl = StringUtils.trimToNull(bundleContext.getProperty(ADAPTIVE_STREAMING_URL_KEY));
-      String readAdaptiveStreamingPort = StringUtils.trimToNull(bundleContext.getProperty(ADAPTIVE_STREAMING_PORT_KEY));
+  public void activate(BundleContext bundleContext, Map<String, Object> properties)
+          throws ComponentException, ConfigurationException {
+    modified(bundleContext, properties);
+  }
 
-      try {
-        streamingUri = getStreamingUrl(readStreamingUrl, readStreamingPort, validStreamingSchemes,
-                DEFAULT_STREAMING_SCHEME, DEFAULT_STREAMING_URL);
-        logger.info("Streaming URL set to \"{}\"", streamingUri);
-      } catch (URISyntaxException e) {
-        throw new ComponentException(String.format("Streaming URL %s could not be parsed", readStreamingUrl), e);
-      }
+  public void modified(BundleContext bundleContext, Map<String, Object> properties)
+          throws ComponentException, ConfigurationException {
 
-      try {
-        adaptiveStreamingUri = getStreamingUrl(readAdaptiveStreamingUrl, readAdaptiveStreamingPort,
-                validAdaptiveStreamingSchemes, DEFAULT_ADAPTIVE_STREAMING_SCHEME, null);
-        logger.info("Adaptive streaming URL set to \"{}\"", adaptiveStreamingUri);
-      } catch (URISyntaxException e) {
-        throw new ComponentException(
-                String.format("Adaptive Streaming URL %s could not be parsed", readAdaptiveStreamingUrl), e);
-      } catch (IllegalArgumentException e) {
-        logger.info("Adaptive streaming URL was not defined in the configuration file");
-      }
+    // get configuration
+    if (properties != null && bundleContext != null) {
 
-      if (adaptiveStreamingUri == null && streamingUri == null) {
-        throw new ComponentException("Streaming URL and adaptive streaming URL are undefined.");
-      }
+      streamingUrls = new HashMap<>();
```

Comment:
1. this should be a ConcurrentHashMap
2. You shouldn't set it first and then fill it, but fill it  and only at the end update the whole configuration at once

Reply:
Will do.